internal/s/error.go

```diff
@@ -422,29 +422,35 @@ type RestartToleranceReached struct {
 	failedChildName        string
 	failedChildErrCount    uint32
 	failedChildErrDuration time.Duration
-	err                    error
+	sourceErr              error
+	lastErr                error
 }
 
 // NewRestartToleranceReached creates an ErrorToleranceReached record
 func NewRestartToleranceReached(
 	tolerance restartTolerance,
-	err error,
-	ch c.Child,
+	sourceCh c.Child,
+	sourceErr error,
+	lastErr error,
 ) *RestartToleranceReached {
 	return &RestartToleranceReached{
-		failedChildName:        ch.GetRuntimeName(),
-		failedChildErrCount:    tolerance.MaxRestartCount,
+		failedChildName: sourceCh.GetRuntimeName(),
+		// we want to indicate that MaxRestartCount is less than the
+		// current error count, given this, we need to increment this
+		// number by one
```

Comment:
yes, my main grief is the fact that the error message says that it failed `n` times on a window `x` of time. 

That `n` was inaccurate due to the fact that it needs to be `n + 1` in order to exceed the tolerance threshold. I can do a simpler change that involves changing the message to

> the supervisor failed more than `n` times 

and that way, I do not need to do this pesky `+1`. 

Thoughts?

Reply:
So, it could have been more than n, say, `n+2`, or is it guaranteed to be `n+1`. I do not want to get too wrapped around the axel here, but `CheckTolerance()` could return the actual value. 

I would do the simplest of what you suggest, and note the `n` that was exceeded, or move the `+1` to `monitor.go:64`. You can ignore my other confused comments.

Reply:
I'll update the message, is easier than doing pesky `+1` arithmethic for the sake of having an error message that makes sense. Thanks for the focus on this.

Reply:
Perfect